naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/GraphQLController.scala

```diff
@@ -112,53 +112,29 @@ class GraphQLController @Inject() (
       variables: JsObject,
       operation: Option[String]): Future[OutgoingQuery] = {
     Future {
-      val startSangriaQueryParser = System.nanoTime()
       QueryParser.parse(query) match {
         case Success(queryAst) =>
-          metricsCollector.markSangriaQueryParserTime(System.nanoTime() - startSangriaQueryParser)
           val baseFilter: IncomingQuery => Future[OutgoingQuery] = (incoming: IncomingQuery) => {
-            val startNaptimeQueryParser = System.nanoTime()
-            SangriaGraphQlParser.parse(query, variables, requestHeader).map { request =>
-              metricsCollector
-                .markNaptimeQueryParserTime(System.nanoTime() - startNaptimeQueryParser)
-              val startEngineExecution = System.nanoTime()
-              val fetcherExecution: Future[Option[Response]] =
-                if (query.contains("IntrospectionQuery")) {
-                  Future.successful(None)
-                } else {
-                  engine.execute(request).map(Some(_))
-                }
-              fetcherExecution.flatMap { responseOpt =>
-                metricsCollector.markEngineExecutionTime(System.nanoTime() - startEngineExecution)
-                val response = responseOpt.getOrElse(Response.empty)
-                val context = SangriaGraphQlContext(response)
-                val startSangriaHydration = System.nanoTime()
-                Executor.execute(
-                  graphqlSchemaProvider.schema,
-                  queryAst,
-                  context,
-                  variables = variables,
-                  exceptionHandler = GraphQLController.exceptionHandler(logger))
-                  .map { executionResponse =>
-                    metricsCollector.markSangriaHydrationTime(
-                      System.nanoTime() - startSangriaHydration)
-                    OutgoingQuery(executionResponse.as[JsObject], Some(response))
-                  }
-              }.recover {
-                case error: QueryAnalysisError =>
-                  OutgoingQuery(error.resolveError.as[JsObject], None)
-                case error: ErrorWithResolver =>
-                  OutgoingQuery(error.resolveError.as[JsObject], None)
-                case error: Exception =>
-                  logger.error("GraphQL execution error", error)
-                  OutgoingQuery(Json.obj("errors" -> Json.arr(error.getMessage)), None)
+            val context = SangriaGraphQlContext(Response.empty, fetcher, requestHeader, ec)
```

Comment:
Eventually, although the actual metrics that are collected will need to be very different now.